fix: fix crash when server-url contains api

Birkbjo opened the pull request:
Part of fix for https://dhis2.atlassian.net/browse/DHIS2-16237
Maintenance app needs to update to new D2 version once this is merged and released for v30. 

Previously the maintenance app would crash when hosted on a server domain that contains "api", eg. `dhis2-api.org`. Because this logic simply replaced the first occurrence. The correct way would be to replace the last occurrence, because we don't want to interfere with the user-definable portion of the url (eg. domain and server context). 

Opted to do the matching programatically instead of replace with regex, because a regex for last occurrence gets pretty complicated and messy.

Birkbjo committed:
fix: fix crash when server-url contains api

tomzemp reviewed (approved):
Looks good to me (and the logic is easier to follow than the existing) ✨

The only (possible but very unlikely) edge case that I could think of where it still might fail is if someone hosted at something like` www.myInstance.org/api/88/lastPartOfURL`, causing the base URL with the actual api /{version} to be `www.myInstance.org/api/88/lastPartOfURL/api/41`

I guess that could be solved by a negative lookahead in the apiUrlWithVersionRexExp, such that the last occurrence of api\/[1-9][0-9] will be selected? But I think that's unlikely to happen, so I don't think it matters to handle.

Birkbjo commented:
> The only (possible but very unlikely) edge case that I could think of where it still might fail is if someone hosted at something like www.myInstance.org/api/88/lastPartOfURL, causing the base URL with the actual api /{version} to be www.myInstance.org/api/88/lastPartOfURL/api/41

Yeah, you're right. I guess we could do this without negative lookahead, by adding `global`-flag and `matchAll` then just selecting the last-match there as well, right?

```js
const baseUrlVersion = baseUrlVersionMatch[baseUrlVersionmatch.length -1]
```

EDIT:

Nvm, I just implemented a negative lookahead and it seems to work. Added a few more test-cases as well.

Birkbjo committed:
fix: support more domain-edge cases

Birkbjo committed:
test: update tests
